src/physfs_platform_libretro.c

```diff
@@ -0,0 +1,421 @@
+    BAIL_IF(opaque == NULL, PHYSFS_ERR_INVALID_ARGUMENT, /* void */);
+    physfs_platform_libretro_vfs->close((struct retro_vfs_file_handle *) opaque);
+} /* __PHYSFS_platformClose */
+
+
+int __PHYSFS_platformDelete(const char *path)
+{
+    BAIL_IF(physfs_platform_libretro_vfs == NULL || physfs_platform_libretro_vfs->remove == NULL, PHYSFS_ERR_NOT_INITIALIZED, 0);
+    return physfs_platform_libretro_vfs->remove(path) == 0 ? 1 : 0;
+} /* __PHYSFS_platformDelete */
+
+
+static PHYSFS_ErrorCode errcodeFromErrnoError(const int err)
+{
+    switch (err)
+    {
+        case 0: return PHYSFS_ERR_OK;
+        case EACCES: return PHYSFS_ERR_PERMISSION;
+        case EPERM: return PHYSFS_ERR_PERMISSION;
+        case EDQUOT: return PHYSFS_ERR_NO_SPACE;
+        case EIO: return PHYSFS_ERR_IO;
+        case ELOOP: return PHYSFS_ERR_SYMLINK_LOOP;
+        case EMLINK: return PHYSFS_ERR_NO_SPACE;
+        case ENAMETOOLONG: return PHYSFS_ERR_BAD_FILENAME;
+        case ENOENT: return PHYSFS_ERR_NOT_FOUND;
+        case ENOSPC: return PHYSFS_ERR_NO_SPACE;
+        case ENOTDIR: return PHYSFS_ERR_NOT_FOUND;
+        case EISDIR: return PHYSFS_ERR_NOT_A_FILE;
+        case EROFS: return PHYSFS_ERR_READ_ONLY;
+        case ETXTBSY: return PHYSFS_ERR_BUSY;
+        case EBUSY: return PHYSFS_ERR_BUSY;
+        case ENOMEM: return PHYSFS_ERR_OUT_OF_MEMORY;
+        case ENOTEMPTY: return PHYSFS_ERR_DIR_NOT_EMPTY;
+        default: return PHYSFS_ERR_OS_ERROR;
+    } /* switch */
+} /* errcodeFromErrnoError */
+
+
+static inline PHYSFS_ErrorCode errcodeFromErrno(void)
+{
+    return errcodeFromErrnoError(errno);
+} /* errcodeFromErrno */
+
+
+int __PHYSFS_platformMkDir(const char *path)
+{
+    int rc;
+    BAIL_IF(physfs_platform_libretro_vfs == NULL || physfs_platform_libretro_vfs->mkdir == NULL, PHYSFS_ERR_NOT_INITIALIZED, -1);
+    rc = physfs_platform_libretro_vfs->mkdir(path);
+    BAIL_IF(rc == -1, PHYSFS_ERR_IO, 0); /* Unknown failure */
+    BAIL_IF(rc == -2, PHYSFS_ERR_DUPLICATE, 0); /* Already exists */
+    return 1;
+} /* __PHYSFS_platformMkDir */
+
+
+int __PHYSFS_platformStat(const char *fname, PHYSFS_Stat *st, const int follow)
+{
+    int32_t size;
+    int flags;
+    BAIL_IF(physfs_platform_libretro_vfs == NULL || physfs_platform_libretro_vfs->stat == NULL, PHYSFS_ERR_NOT_INITIALIZED, 0);
+    flags = physfs_platform_libretro_vfs->stat(fname, &size);
+    if (!(flags & RETRO_VFS_STAT_IS_VALID)) {
+        BAIL(PHYSFS_ERR_NOT_FOUND, 0);
+    }
+
+    if (flags & RETRO_VFS_STAT_IS_DIRECTORY) {
+        st->filetype = PHYSFS_FILETYPE_DIRECTORY;
+        st->filesize = 0;
+    } else if (flags & RETRO_VFS_STAT_IS_CHARACTER_SPECIAL) {
+        st->filetype = PHYSFS_FILETYPE_OTHER;
+        st->filesize = 0;
+    } else {
+        st->filetype = PHYSFS_FILETYPE_REGULAR;
+        st->filesize = size;
+    }
+
+    /* TODO: Fill the following properties */
```

Comment:
libretro's virtual file system doesn't actually provide these values, so perhaps we just make a note of it in the comments...

``` suggestion
    /* libretro's virtual file system doesn't provide support for retrieving the following values */
```

Reply:
Most excellent. Returning `-1` for the invalid time values now.